backend/routes/db.py

```diff
@@ -314,49 +325,54 @@ async def get_concept_ids_by_codeset_id_post(request: Request, codeset_ids: Unio
 
 @router.get("/concept-ids-by-codeset-id")
 @return_err_with_trace
-async def get_concept_ids_by_codeset_id(request: Request, codeset_ids: Union[List[str], None] = Query(...)) -> Dict:
+async def get_concept_ids_by_codeset_id(
+    request: Request, codeset_ids: Union[List[str], None] = Query(...)
+) -> Dict[str, str]:
+    """Get concept IDs by codeset id"""
+    q = f"""
+          SELECT csids.codeset_id, COALESCE(cibc.concept_ids, ARRAY[]::integer[]) AS concept_ids
+          FROM (VALUES{",".join([f"({csid})" for csid in codeset_ids])}) AS csids(codeset_id)
+          LEFT JOIN concept_ids_by_codeset_id cibc ON csids.codeset_id = cibc.codeset_id"""
     if not codeset_ids:
-        return [[]]
+        return {}
 
     rpt = Api_logger()
     await rpt.start_rpt(request, params={'codeset_ids': codeset_ids})
 
-    with get_db_connection() as con:
-        try:
-            q = f"""
-                  SELECT csids.codeset_id, COALESCE(cibc.concept_ids, ARRAY[]::integer[]) AS concept_ids
-                  FROM (VALUES{",".join([f"({csid})" for csid in codeset_ids])}) AS csids(codeset_id)
-                  LEFT JOIN concept_ids_by_codeset_id cibc ON csids.codeset_id = cibc.codeset_id"""
+    try:
+        with get_db_connection() as con:
```

Comment:
## Type 1/3: `with` block
There were many instances of offending `with` blocks, where they would either (a) return within the block, or there was (b) a try/except within the block. I saw both such examples in the current log of idle connections in the database.

There were quite a lot of instances of 'a'.